lib/ansible/plugins/__init__.py

```diff
@@ -52,13 +52,24 @@ class AnsiblePlugin(ABC):
     def __init__(self):
         self._options = {}
 
+    def _handle_deprecations(self):
+        ''' assumes that any generated deprecats have been 'locally' sourced '''
+
+        for deprecation in C.config.DEPRECATED[:]:
+            msg =  "{0}'s {1} {2}. Alternatives: {2}".format(self._load_name, deprecation[0], deprecation[1]['why'], deprecation[1]['alternatives'])
+            display.deprecated(msg, version=deprecation[1]['version'])
+            C.config.DEPRECATED.remove(deprecation)
```

Comment:
Can it happen that multiple threads run this in parallel?

Reply:
forks, not threads, yes, that is why i made that dependency to the other PR (which proxies display to parent process, which does serialization and de duping)